docs/writing-rules/data-flow/overview.md

```diff
@@ -5,22 +5,36 @@ description: >-
+# Data-flow analysis engine overview
+
+Semgrep provides an intra-procedural data-flow analysis engine that provides it with additional capabilities. [Constant propagation](../constant-propagation/) allows Semgrep to, for example, match `return 42` against `return x` when `x` can be reduced to `42` by constant folding. [Taint tracking (aka taint mode)](../taint-mode/) enables you to write simple rules that catch complex [injection bugs](https://owasp.org/www-community/Injection_Flaws), such as those that can result in [cross-site scripting (XSS)](https://owasp.org/www-community/attacks/xss/).
 
 In principle, all data-flow related features are available for any of our [supported languages](/supported-languages/). For further details please check the [data-flow status documentation](../status/). 
 
 Make sure that you understand the [design trade-offs](#design-trade-offs) and limitations of the data-flow engine.
 
 At present, Semgrep provides no user-friendly way of specifying your own data-flow analyses, but please [let us know if you have suggestions](https://github.com/returntocorp/semgrep/issues/new/choose). And, if you can code in OCaml, then you are very welcome to [contribute](/docs/contributing/contributing/).
 
+Data-flow analysis is also supported in DeepSemgrep. For more details, see [DeepSemgrep](/docs/deepsemgrep/) documentation.
+
 ## Design trade-offs
 
 Semgrep strives for simplicity and to deliver on the promise of lightweight and fast static analysis. In addition to being intra-procedural, here are some other trade-offs:
 
 - No path sensitivity: All _potential_ execution paths are considered, despite that some may not be feasible.
-- No pointer or shape analysis: _Aliasing_ that happens in non-trivial ways may not be detected, such as through arrays or pointers. Individual elements in arrays or other data structures are not tracked. There is also no proper field sensitivity at present, but this may be developed in the future.
+- No pointer or shape analysis: _Aliasing_ that happens in non-trivial ways may not be detected, such as through arrays or pointers. TODO Individual elements in arrays or other data structures are not tracked. There is also no proper field sensitivity at present, but this may be developed in the future.
 - No soundness guarantees: Semgrep ignores the effects of `eval`-like functions on the program state. It doesn’t make worst-case sound assumptions, but rather "reasonable" ones.
```

Comment:
I am not sure what this whole list means. We are trying to list very specific programmatic limitations. Are they still valid? Not sure about this whole thing. Maybe @IagoAbal can help here?

Reply:
The limitations still apply (and will likely apply for the foreseeable future) and advanced users need to know about them.